### PR TITLE
[GH37] Slim down options on the repository page

### PR DESCRIPTION
**Pull request by timallen527**

### CODE REVIEW

File: lib/static/static/js/controllers/repositoriesCtrl.js [hunk 1]
```diff
@@ -1,29 +1,67 @@
 (function() {
-  angular.module('huburn').controller('repositoriesCtrl', ['$scope', '$http', 'navigationService', function($scope, $http, navigationService) {
-    $scope.repositories = [];
+  angular
+    .module('huburn')
+    .controller('repositoriesCtrl', repositoriesCtrl);
+
+  repositoriesCtrl.$inject = ['$scope', '$http', 'navigationService'];
```
Comment (cidthecoatrack) [inline lib/static/static/js/controllers/repositoriesCtrl.js]: Thanks for the refactor

File: lib/static/static/js/controllers/repositoriesCtrl.js [hunk 2]
```diff
@@ -1,29 +1,67 @@
...
+  function repositoriesCtrl($scope, $http, navigationService) {
+    
+    $scope.repositories = [];
     navigationService.clear();
 
-    $http.get('/github', { params: { path: '/user/repos' } }).success(function(repos) {
-      addRepos(repos);
-    });
+    var addUserRepos = function() {
+      $http.get('/github', { params: { path: '/user/repos' } }).success(function(repos) {
```
Comment (cidthecoatrack) [inline lib/static/static/js/controllers/repositoriesCtrl.js]: There is a githubService you can use that does the $http parsing.  See the burnupCtrl for examples on how to use it.

File: lib/static/static/js/controllers/repositoriesCtrl.js [hunk 3]
```diff
@@ -1,29 +1,67 @@
...
+        });
+      });
+    };
 
-    $http.get('/github', { params: { path: '/user/orgs' } }).success(function(orgs) {
-      orgs.forEach(function(org) {
-        $http.get('/github', { params: { path: '/orgs/' + org.login + '/repos' } }).success(function(repos) {
-          addRepos(repos);
+    var addOrgRepos = function() {
+      $http.get('/github', { params: { path: '/user/orgs' } }).success(function(orgs) {
```
Comment (cidthecoatrack) [inline lib/static/static/js/controllers/repositoriesCtrl.js]: Again, ditto the github service

File: lib/static/static/js/controllers/repositoriesCtrl.js [hunk 4]
```diff
@@ -1,29 +1,67 @@
...
+          addFilteredOrgRepos(org);
         });
       });
-    });
+    };
 
-    var addRepos = function(repos) {
-      var newRepos = $scope.repositories.concat(repos);
+    var addFilteredOrgRepos = function(org) {
+      $http.get('/github', { params: { path: '/orgs/' + org.login + '/repos' } }).success(function(repos) {
```
Comment (cidthecoatrack) [inline lib/static/static/js/controllers/repositoriesCtrl.js]: github service

File: lib/static/static/js/controllers/repositoriesCtrl.js [hunk 5]
```diff
@@ -1,29 +1,67 @@
...
+        repos.forEach(function(repo) { 
+          addRepoWithAnyMilestoneThatHasAssociatedIssues(repo);
+        });
+      });
+    };
+
+    var addRepoWithAnyMilestoneThatHasAssociatedIssues = function(repo) {
+      var milestonesPath = '/repos/' + repo.owner.login + '/' + repo.name + '/milestones';
+      $http.get('/github', { params: { path: milestonesPath } }).success(function(milestones) {
```
Comment (cidthecoatrack) [inline lib/static/static/js/controllers/repositoriesCtrl.js]: github service

File: lib/static/static/js/controllers/repositoriesCtrl.js [hunk 6]
```diff
@@ -1,29 +1,67 @@
...
+        if (anyMilestoneHasAssociatedIssues(milestones))
+          addRepo(repo);
+      });
+    }
+
+    var anyMilestoneHasAssociatedIssues = function(milestones) {
+      return milestones.some(function(milestone) {
+        if (milestone.open_issues > 0 || milestone.closed_issues > 0)
```
Comment (cidthecoatrack) [inline lib/static/static/js/controllers/repositoriesCtrl.js]: isn't this the same as "return (milestone.open_issues > 0 || milestone.closed_issues > 0);" ?  Also, good find with the ".some" function.  Do we have to check both open and closed, or can we just see something like "issues" or "total_issues"?

**Comment by efouts** [comment]
I get this js error when opening the repositories view. It looks like were calling success rather than then.

TypeError: undefined is not a function
    at addUserRepos (https://localhost:8443/js/controllers/repositoriesCtrl.js:15:10)
    at new repositoriesCtrl (https://localhost:8443/js/controllers/repositoriesCtrl.js:67:5)
    at e (https://localhost:8443/js/vendor/angular.min.js:36:94)
    at Object.instantiate (https://localhost:8443/js/vendor/angular.min.js:36:223)
    at $get (https://localhost:8443/js/vendor/angular.min.js:69:253)
    at link (https://localhost:8443/js/vendor/angular-route.min.js:7:248)
    at B (https://localhost:8443/js/vendor/angular.min.js:55:414)
    at g (https://localhost:8443/js/vendor/angular.min.js:48:319)
    at https://localhost:8443/js/vendor/angular.min.js:47:445
    at $get.h (https://localhost:8443/js/vendor/angular.min.js:49:266) <div class="container body-content ng-scope" ng-view="">

**Comment by timallen527** [comment]
@efouts This should be corrected now. Thanks.

**Comment by efouts** [comment]
What do we think about the request errors, 410s, for repos with issues disabled? I get 13 errors in my console for example. Is there a way to handle those so they do no show as errors in the browser? An example one is below that we all should see when running locally. 

GET https://localhost:8443/github?path=%2Frepos%2FQualtrax%2FQualtraxCloud%2Fmilestones 410 (Gone)

/cc @jramey @cidthecoatrack

**Comment by timallen527** [comment]
@efouts Yeah, I was seeing those as well. I'm not sure if we can filter those out so that we don't make the requests for those with that criteria, but I'll see what I can dig up.

**Comment by timallen527** [comment]
@efouts I have pushed a fix for this. As it turns out, there is a property on the repo called "has_issues" which actually indicates whether issues are turned on or off for the repo. So, I used that.

**Comment by efouts** [comment]
:+1:

**Pull request merged**